airgun/entities/cloud_inventory.py

```diff
@@ -19,27 +17,55 @@ def read(self, entity_name=None, widget_names=None):
             result.update(view.inventory_list.read())
         return result
 
+    def configure_cloud_connector(self):
+        """Configure Cloud Connector"""
+        view = self.navigate_to(self, 'All')
+        view.cloud_connector.click()
+        wait_for(
+            lambda: view.cloud_connector_status.is_displayed is True,
+            handle_exception=True,
+            timeout=20,
+            delay=1,
+            logger=view.logger,
+        )
+        wait_for(
+            lambda: view.cloud_connector_status.is_displayed is False,
+            handle_exception=True,
+            timeout=180,
+            delay=1,
+            logger=view.logger,
+        )
```

Comment:
Until we have a good resolution for the `ensure_page_safe` default timeout issue (https://github.com/SatelliteQE/airgun/issues/578), I would suggest adding calls to that method instead of `sleep()` or extra `wait_for()` calls.

E.g., after the `click()`, `view.browser.plugin.ensure_page_safe(timeout='60s')` would wait up to 60s for the spinner and other script / component activity to complete. `ensure_page_safe()` is called automatically before every element lookup, but the default timeout of `30s` isn't enough here.

Reply:
@tpapaioa I tried using `view.browser.plugin.ensure_page_safe(timeout='60s')` and I see no issue in using `configure_cloud_connector` and `sync_inventory_status`. But for `generate_report` and `download_report` I am facing the same issue. i.e it's not waiting for `generate_report` to finish first and goes directly to upload tab.

Reply:
@tpapaioa @mshriver what should we do? should we keep `wait_for`?

Reply:
Use whatever logic most accurately determines whether the action has completed successfully. I doubt that waiting for spinners or buttons to change state temporarily, and then wait again for them to change a second time, is the best strategy. In the case of clicking `Configure Cloud Connector`, for example, it starts a job task, so waiting on the completion of that job and returning its output status would probably be the most reliable method. In the case of report generation and upload, there are foreman/dynflow tasks that you can wait on.

Reply:
@tpapaioa makes sense. I completely forgot that we could use `wait_for_tasks` in our tests itself instead of relying on airgun. 
I will simply update my test and wait for tasks to finish.
like this:
```
timestamp = (datetime.utcnow() - timedelta(minutes=2)).strftime('%Y-%m-%d %H:%M')
wait_for_tasks(
    search_query=(
        'label = InventorySync::Async::InventoryFullSync'
        f' and started_at >= "{timestamp}"'
    ),
    search_rate=15,
    max_tries=10,
)
wait_for_tasks(
    search_query=(
        'Remote action: Configure cloud connector'
        f' and started_at >= "{timestamp}"'
    ),
    search_rate=15,
    max_tries=10,
)
wait_for_tasks(
    search_query='label = ForemanInventoryUpload::Async::GenerateReportJob'
                 f' and started_at >= "{timestamp}"',
    search_rate=15,
    max_tries=10,
)
```

Reply:
update the pr.